src/utils/framePostHandler.ts

```diff
@@ -47,20 +61,76 @@ export async function framePostHandler(req: NextApiRequest) {
   const headers = new Headers();
   headers.append('Content-Type', 'text/html');
 
-  return new Response(
-    `
-      <html>
-        <meta property="fc:frame" content="vNext">
-        ${hasPrevious ? '<meta property="fc:frame:button:1" content="←">' : ''}
-        <meta property="fc:frame:button:${hasPrevious ? 2 : 1}" content="→">
-        <meta property="fc:frame:image" content="${url}">
-        <meta property="fc:frame:post_url" content="${url}">
-        <body>gm</body>
-      </html>
-    `,
-    {
-      status: 200,
-      headers,
+  const showTwoButtons = hasPrevious;
+  const frameButtons: [FrameButtonMetadata, ...FrameButtonMetadata[]] = [
+    { label: showTwoButtons ? '←' : buttonContent, action: 'post' },
+    ...(showTwoButtons ? [{ label: buttonContent, action: 'post' } as FrameButtonMetadata] : []),
+  ];
+  const image = url.toString();
+  const postUrl = url.toString();
+
+  const htmlConfig = {
+    buttons: frameButtons,
+    image: {
+      src: image,
+      aspectRatio: '1.91:1' as AllowedAspectRatio,
+    },
+    postUrl,
+  };
+
+  position = url.searchParams.get('position');
+
+  // use square aspect ratio for image if appropriate for collection token
+  // TODO(rohan): similarly support it for other types of frames
+  if (handleSquareAspectRatioType === 'CollectionFrame' && position) {
+    let squareAspectRatio = false;
+
+    const chain = url.searchParams.get('chain');
+    const contractAddress = url.searchParams.get('contractAddress');
+
+    if (!chain || typeof chain !== 'string') {
```

Comment:
would this cause an error for collection page urls that dont include a chain? like https://gallery.so/artblocks/0xa7d8d9ef8d8ce8992df33d8b8cf4aebabd5bd270/304

Reply:
yes, left a comment above to follow up later!